src/material-experimental/mdc-checkbox/harness/checkbox-harness.spec.ts

```diff
@@ -0,0 +1,191 @@
+import {HarnessLoader} from '@angular/cdk-experimental/testing';
+import {TestbedHarnessEnvironment} from '@angular/cdk-experimental/testing/testbed';
+import {Component} from '@angular/core';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {FormControl, ReactiveFormsModule} from '@angular/forms';
+import {MatCheckboxModule} from '@angular/material/checkbox';
+import {MatCheckboxModule as MatMdcCheckboxModule} from '../index';
+import {MatCheckboxHarness} from './checkbox-harness';
+import {MatCheckboxHarness as MatMdcCheckboxHarness} from './mdc-checkbox-harness';
+
+let fixture: ComponentFixture<unknown>;
+let loader: HarnessLoader;
+let checkboxHarness: typeof MatCheckboxHarness;
+
+describe('MatCheckboxHarness', () => {
+  describe('non-MDC-based', () => {
+    beforeEach(async () => {
+      await TestBed.configureTestingModule({
+        imports: [MatCheckboxModule, ReactiveFormsModule],
+        declarations: [CheckboxHarnessTest],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(CheckboxHarnessTest);
+      fixture.detectChanges();
+      loader = TestbedHarnessEnvironment.loader(fixture);
+      checkboxHarness = MatCheckboxHarness;
+    });
+
+    runTests();
+  });
+
+  describe('MDC-based', () => {
+    beforeEach(async () => {
+      await TestBed.configureTestingModule({
+        imports: [MatMdcCheckboxModule, ReactiveFormsModule],
+        declarations: [CheckboxHarnessTest],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(CheckboxHarnessTest);
+      fixture.detectChanges();
+      loader = TestbedHarnessEnvironment.loader(fixture);
+      // Public APIs are the same as MatCheckboxHarness, but cast is necessary because of different
+      // private fields.
+      checkboxHarness = MatMdcCheckboxHarness as any;
+    });
+
+    runTests();
+  });
+});
+
+/** Shared tests to run on both the original and MDC-based checkboxes. */
+function runTests() {
+  it('should load all checkbox harnesses', async () => {
+    const checkboxes = await loader.getAllHarnesses(checkboxHarness);
+    expect(checkboxes.length).toBe(2);
+  });
+
+  it('should load checkbox with exact label', async () => {
+    const checkboxes = await loader.getAllHarnesses(checkboxHarness.with({label: 'First'}));
+    expect(checkboxes.length).toBe(1);
+    expect(await checkboxes[0].getLabelText()).toBe('First');
```

Comment:
Instead of referencing the checkbox instances in this file like `checkboxes[0]`, could we introduces meaningful names for them? Something like
```ts
requiredCheckedCheckbox = checkboxes[0];
disabledIndeterminateCheckbox = checkboxes[1];
```
You could alternatively use multiple aliases for the same instances:
```ts
requiredCheckbox = enabledCheckbox = checkboxes[0];
disabledCheckbox = indeterminateCheckbox = checkboxes[1];
```